airflow/ui/src/components/Graph/DownloadButton.tsx

```
@@ -0,0 +1,83 @@
+ * under the License.
+ */
+import { Button } from "@chakra-ui/react";
+import { Panel, useReactFlow, getNodesBounds, getViewportForBounds } from "@xyflow/react";
+import { toPng } from "html-to-image";
+import { useEffect, useState } from "react";
+import { FiDownload } from "react-icons/fi";
+
+export const DownloadButton = () => {
+  const { getNodes } = useReactFlow();
+  const [dimensions, setDimensions] = useState({ height: 768, width: 1024 });
+
+  useEffect(() => {
+    const updateDimensions = () => {
+      const container = document.querySelector(".react-flow__viewport");
+
+      if (container) {
+        setDimensions({
+          height: container.clientHeight,
+          width: container.clientWidth,
+        });
+      }
+    };
+
+    updateDimensions();
+    globalThis.addEventListener("resize", updateDimensions);
+
+    return () => globalThis.removeEventListener("resize", updateDimensions);
+  }, []);
+
+  const onClick = () => {
+    const nodesBounds = getNodesBounds(getNodes());
+    const viewport = getViewportForBounds(nodesBounds, dimensions.width, dimensions.height, 0.5, 2, 0.25);
+    // Method obtained from https://reactflow.dev/examples/misc/download-image
+    const container = document.querySelector(".react-flow__viewport");
+
+    if (container instanceof HTMLElement) {
+      toPng(container, {
+        height: dimensions.height,
+        style: {
+          height: `${dimensions.height}px`,
+          transform: `translate(${viewport.x}px, ${viewport.y}px) scale(${viewport.zoom})`,
+          width: `${dimensions.width}px`,
+        },
+        width: dimensions.width,
+      })
+        .then((dataUrl) => {
+          const downloadLink = document.createElement("a");
+
+          downloadLink.setAttribute("download", "reactflow.png");
+          downloadLink.setAttribute("href", dataUrl);
+          downloadLink.click();
+        })
+        .catch(() => {
+          // void catch
+        });
```

I'm not sure about that, we could at least show a toaster or something. Silently catching all and failing is not great for user. Could be frustrating to just have a button that does nothing.

---

Added toaster from components folder.